create.ps1

```diff
@@ -163,136 +108,166 @@ function Resolve-ZivverError {
+    $headers.Add("Authorization", "Bearer $($actionContext.Configuration.Token)")
+    Write-Verbose "Created authorization headers"
+    #endregion Create authorization headers
+
+    #region Get Zivver account
+    $actionMessage = "querying Zivver account"
+
+    $getZivverSplatParams = @{
+        Headers  = $headers
+        Endpoint = "Users?filter=$correlationField eq ""$correlationValue"""
+        Method   = 'GET'
     }
 
-    Write-Verbose "Verifying if Zivver account for [$($p.DisplayName)] must be created or correlated"
-    $splatParams['Endpoint'] = "Users?filter=userName eq ""$($account.userName)"""
-    $splatParams['Method'] = 'GET'
-    $responseUser = (Invoke-ZivverRestMethod @splatParams).Resources
+    $correlatedAccount = (Invoke-ZivverRestMethod @getZivverSplatParams).Resources
+        
+    Write-Verbose "Queried Ziver account where [$($correlationField)] = [$($correlationValue)]. Result: $($correlatedAccount | ConvertTo-Json)"
+    #endregion Get Zivver account
 
-    if ($responseUser.Length -lt 1) {
-        $action = 'Create-Correlate'
-        $dryRunMessage = "Create Zivver account for: [$($p.DisplayName)], will be executed during enforcement."
+    #region Calulate action
+    $actionMessage = "calculating action"
+    if (($correlatedAccount | Measure-Object).count -eq 0) {
+        $actionAccount = "Create"
     }
-    elseif ($responseUser | Where-Object { $_.userName -eq $account.userName }) {
-        $action = 'Correlate'
-        $dryRunMessage = "Correlate Zivver account for: [$($p.DisplayName)], will be executed during enforcement."
+    elseif (($correlatedAccount | Measure-Object).count -eq 1) {
+        $actionAccount = "Correlate"
+    }
+    elseif (($correlatedAccount | Measure-Object).count -gt 1) {
+        $actionAccount = "MultipleFound"
+    }
+    #endregion Calulate action
 
-        if ($($config.UpdatePersonOnCorrelate -eq "true")) {
-         
-            # When correlate make user active
-            $responseUser | Add-Member -MemberType NoteProperty -Name "active" -Value $account.active -Force
+    #region Process
+    switch ($actionAccount) {
+        "Create" {
+            $actionMessage = "creating account"
+            # Create account with only required fields
 
-            if ($account.'urn:ietf:params:scim:schemas:zivver:0.1:User'.PSObject.Properties.Name -contains 'SsoAccountKey') {
-                $responseUser.'urn:ietf:params:scim:schemas:zivver:0.1:User' | Add-Member -MemberType NoteProperty -Name "SsoAccountKey" -Value $account.'urn:ietf:params:scim:schemas:zivver:0.1:User'.SsoAccountKey -Force
+            $postZivverSplatParams = @{
+                Headers  = $headers
+                Endpoint = 'Users'
+                Method   = 'POST'
+                Body     = $account | ConvertTo-Json
             }
 
-            $responseUser.name.formatted = $account.name.formatted
+            Write-Verbose "SplatParams: $($postZivverSplatParams | ConvertTo-Json)"
```

Comment:
Deze zie ik ook zo af en toe voorbijkomen. Het terugschrijven van de splatParams in een debug message. Is dat iets waarvan jullie zeggen; stop maar in het template?

Reply:
Als je een error krijgt bij een POST/PUT dan kan je met debug toggle aan en retry aan direct zien wat stuurt HelloID nu naar die API. Dit is daarom fijn. Zoals besproken gaan we hier nog verder naar kijken.

Reply:
Voor nu verwijderd